lib/services/elastic-helpers.test.js

```diff
@@ -28,25 +28,43 @@ describe(_.startCase(filename), function () {
       it('returns an array of ops if type property equals "put" with a JSON string', function () {
         var ops = [{ value: '{}', type: 'put' }];
 
-        expect(fn('index', 'type', ops)).to.deep.equal([{ index: { _index: 'index', _type: 'type' } }, {}]);
+        expect(fn({index: 'index', type: 'type', ops: ops})).to.deep.equal([{ index: { _index: 'index', _type: 'type' } }, {}]);
       });
 
       it('returns an array of ops if type property equals "put" with a JS object', function () {
         var ops = [{ value: {}, type: 'put' }];
 
-        expect(fn('index', 'type', ops)).to.deep.equal([{ index: { _index: 'index', _type: 'type' } }, {}]);
+        expect(fn({index: 'index', type: 'type', ops: ops})).to.deep.equal([{ index: { _index: 'index', _type: 'type' } }, {}]);
       });
 
       it('assigns a "key" property if one is defined in the op', function () {
         var ops = [{ value: '{}', type: 'put', key: 'key' }];
 
-        expect(fn('index', 'type', ops)).to.deep.equal([{ index: { _id: 'key', _index: 'index', _type: 'type' } }, {}]);
+        expect(fn({index: 'index', type: 'type', ops: ops})).to.deep.equal([{ index: { _id: 'key', _index: 'index', _type: 'type' } }, {}]);
       });
 
       it('assigns a "key" property if one is defined in the op', function () {
         var ops = [{ value: '{}', type: 'get', key: 'key' }];
 
-        expect(fn('index', 'type', ops)).to.deep.equal([]);
+        expect(fn({index: 'index', type: 'type', ops: ops})).to.deep.equal([]);
+      });
+
+      it('allows for update', function () {
+        var ops = [{ value: '{}', type: 'put', key: 'key' }];
+
+        expect(fn({index: 'index', type: 'type', ops: ops, action: 'update'})).to.deep.equal([{ update: { _id: 'key', _index: 'index', _type: 'type' } }, {doc: {}}]);
+      });
+
+      it('allows for delete', function () {
+        var ops = [{ value: '{}', type: 'put', key: 'key' }];
+
+        expect(fn({index: 'index', type: 'type', ops: ops, action: 'delete'})).to.deep.equal([{ delete: { _id: 'key', _index: 'index', _type: 'type' } }, {}]);
```

Comment:
`delete` batch ops have no following source object because a `delete` is the complete operation